### PR TITLE
dev/user-interface#60 Fixes UI regression & makes more accordions accessible

### PR DESCRIPTION
**Pull request by vingle**
While testing @aydun's https://github.com/civicrm/civicrm-core/pull/29447, I discovered a CSS-based UI regression from the CSS changes in 5.69:  accordions using the `.crm-collapsible` pattern (of which I've found 9) don't change the expand/close caret icon on open close. The accordions still work, but the triangle is set to open position, even when the accordion is closed.

Rather than just fix this with CSS, I figured this was a good chance to replace the pattern with the more accessible markup pattern. Between this and Aidan's PR, this would only leave a handful of non-accessible accordions left.

Overview
----------------------------------------
I've gone thru all but one of the instances of `.crm-collapsible` for accordions and replaced with `<details><summary>` using `.crm-accordion-light` where appropriate to try to match the original UI. Have tested each, and checked for related JS in the template files that might now error out.

Before
----------------------------------------
image

After
----------------------------------------
image

Technical details
----------------------------------------
I've added container `<div class="crm-accordion-body">` where that didn't exist. In one instance I've moved a region that was ins

Comments
----------------------------------------
Testing template files can take a bit of detective work to find them in the UI, so here's the paths for each:

- [templates/CRM/Admin/Form/ScheduleReminders.tpl](https://github.com/civicrm/civicrm-core/compare/master...vingle:civicrm-core:accordion-collapsible?expand=1#diff-ddb3dcbc67957dd83238c306747e5a7aec7571226b43498b231bcd50afb1b0a2) - civicrm/admin/scheduleReminders?reset=1 & click 'add reminder'.
- [templates/CRM/Admin/Page/APIExplorer.tpl](https://github.com/civicrm/civicrm-core/compare/master...vingle:civicrm-core:accordion-collapsible?expand=1#diff-9bc5ef28509d482da523fbcd9d4f4e67bdb9161ca5ad5f7d92411ec30a142a07) - civicrm/api3 & then select an Entity, such as ACLRole.
- [templates/CRM/Case/Form/ActivityTab.tpl](https://github.com/civicrm/civicrm-core/compare/master...vingle:civicrm-core:accordion-collapsible?expand=1#diff-53b29c81373de6cd571173e44d11e0fcff6eb820d33e40362ac6805716715dd7) - civicrm/contact/view/case?reset=1 under Activities/SearchFilters (might need a case) - NB despite all the red/green, I just changed two different accordion patterns here, added a body wrapper and adjusted the tabbing, but the diff looks much bigger than that
- [templates/CRM/Contact/Form/Edit/Address/CustomData.tpl](https://github.com/civicrm/civicrm-core/compare/master...vingle:civicrm-core:accordion-collapsible?expand=1#diff-e43d9ec9e1d9ed39a08acd5306f17cd44eb56d9af7457ab08656ca9ea4002d0c) - civicrm/contact/add?reset=1 + the address fields need custom data
- [templates/CRM/Contact/Form/Edit/Email.tpl](https://github.com/civicrm/civicrm-core/compare/master...vingle:civicrm-core:accordion-collapsible?expand=1#diff-49780f8e22b1a913a8bafb47ace49cec0da54b475b0998acfa266118ae887c4a) - civicrm/contact/add?reset=1 the primary email (after save) should have an accordion to add a signature.
- [templates/CRM/Contribute/Form/ManagePremiums.tpl](https://github.com/civicrm/civicrm-core/compare/master...vingle:civicrm-core:accordion-collapsible?expand=1#diff-8c05f99d5ba00c9a41275653a3df8005bca23ac2d9eb18bea435ee65143dbbe7) - civicrm/admin/contribute/managePremiums?reset=1 + click 'add premium'.
- [templates/CRM/Event/Form/ManageEvent/Registration.tpl](https://github.com/civicrm/civicrm-core/compare/master...vingle:civicrm-core:accordion-collapsible?expand=1#diff-ee2d128392f43a5bf8e331ebef8ac5c01f53bd1907b2159619a5b7553ab4aa01) - civicrm/event/manage/settings?reset=1&action=update&id=2&selectedChild=registration - the large diff here is down to wrapping a chunk in a `crm-accordion-body` div.
- [templates/CRM/Financial/Form/FinancialBatch.tpl](https://github.com/civicrm/civicrm-core/compare/master...vingle:civicrm-core:accordion-collapsible?expand=1#diff-f6f98a47495efd220aefcc9ae71f387b94cfbc0d4c1fb2b44303d67135a0a273) - civicrm/financial/batch?reset=1&action=add

### CODE REVIEW

**Comment by highfalutin** [comment]
Currently reviewing; tracking manual testing at https://lab.civicrm.org/dev/user-interface/-/issues/67#testing-29533

**Comment by vingle** [comment]
One note when reviewing this: @colemanw fixed one of these last week (https://github.com/civicrm/civicrm-core/pull/29446) and decided in the end it didn't need to be an accordion at all as so little was revealed – that's an alternative path with these.

**Comment by shaneonabike** [comment]
Hi Nicol,

Quick question: I noticed that for custom_groups in Event Registration it is not converting the pre-existing divs to a details/summary combo. I tried manually changing it and things work. I just wanted to checkin about that (and perhaps it's just in another PR?)

Selection_010

cc @vingle

**Comment by vingle** [comment]
@shaneonabike - If I remember right, I think custom fields are being handled in #29448 in these files:
- https://github.com/civicrm/civicrm-core/pull/29448/files#diff-aea937f7f4ec3ca3876c8050f77dccd235a352df45213d3bab2f518174d9c96c
- https://github.com/civicrm/civicrm-core/pull/29448/files#diff-aea937f7f4ec3ca3876c8050f77dccd235a352df45213d3bab2f518174d9c96c

**Comment by shaneonabike** [comment]
Good to go from Wordpress perspective extra for the extra class you added causing theming issues. If you fix that I'm good on this point to merge.

cc @vingle

**Comment by highfalutin** [comment]
I reviewed the first three changes at https://lab.civicrm.org/dev/user-interface/-/issues/67#testing-29533. The API v3 Explorer change breaks the interface, so IMO we should remove that one. The Case View one creates some extra (kind of ugly) borders around the detail content. I'll try to make a commit that fixes these.

**Comment by vingle** [comment]
Thanks alot @highfalutin - I hadn't seen the API3 issues.

The css border was on my radar and I was waiting for the accordions to be merged before putting in a css tidy-up. But there's no harm in making that PR now: https://github.com/civicrm/civicrm-core/pull/29551. It fixes both the border that is mistakenly inherited from `.crm-accordion-bold` and adds a tiny bit of breathing space between multiple accordions, similar to how it was previously.

**Comment by vingle** [comment]
> Good to go from Wordpress perspective extra for the extra class you added causing theming issues. If you fix that I'm good on this point to merge.

@shaneonabike - are you able to give a bit more details/screengrab? The new classes are in civicrm.css, which WordPress loads, is this specific to CAU? Quite [a few of the new classes](https://github.com/christianwach/civicrm-admin-utilities/blob/master/assets/css/civicrm-admin-utilities-admin-civi-5-69-plus.css) have been added there by Christian.

**Comment by highfalutin** [comment]
@vingle @herbdool thanks for fixing the border and spacing issues! I'm about to respond to @vingle's feedback on my [PR 13](https://github.com/vingle/civicrm-core/pull/13).

**Comment by highfalutin** [comment]
So at this point, I've reviewed and approved/submitted fixes for the first 3 files; Shane has reviewed and approved the rest. Once Nic merges my changes ([13](https://github.com/vingle/civicrm-core/pull/13)), this PR can be merged.

**Review by shaneonabike — CHANGES_REQUESTED** [review]

**Comment by shaneonabike** [comment]
> > Good to go from Wordpress perspective extra for the extra class you added causing theming issues. If you fix that I'm good on this point to merge.
> 
> @shaneonabike - are you able to give a bit more details/screengrab? The new classes are in civicrm.css, which WordPress loads, is this specific to CAU? Quite [a few of the new classes](https://github.com/christianwach/civicrm-admin-utilities/blob/master/assets/css/civicrm-admin-utilities-admin-civi-5-69-plus.css) have been added there by Christian.

No I wrote a review point but it was still in pending -- I didn't realize I needed to submit it - :scream: . Anyway you'll need the slight change. Other than that you are totally good from my testing on WP.

cc @vingle

File: templates/CRM/Contact/Form/Edit/Email.tpl
Comment (vingle) [reply templates/CRM/Contact/Form/Edit/Email.tpl:36]: Thanks @shaneonabike. That misplaced border should vanish with two other PRs: the CSS tweak that's merged (#29551) and the changes to the contact edit screen accordions dropping the `.crm-accordion-wrapper` class in #29448.

Comment (vingle) [reply templates/CRM/Contact/Form/Edit/Email.tpl:36]: NB - the reason why I'm loathe to fix these by removing the `.crm-accordion-body` class is it's the simplest method to set a Civi wide padding value for accordion contents.

**Comment by vingle** [comment]
Thanks @highfalutin, @shaneonabike & @herbdool – the API3 fix is merged, and the email signature issue should be fixed when #29448 (or the part of it relating to the contact page, which has passed tests) is merged.

File: templates/CRM/Contact/Form/Edit/Email.tpl
Comment (highfalutin) [reply templates/CRM/Contact/Form/Edit/Email.tpl:36]: Ok, I think this is resolved then.

**Comment by highfalutin** [comment]
@colemanw test failures are unrelated since this pr only touches `. tpl` files

File: templates/CRM/Admin/Form/ScheduleReminders.tpl
Comment (colemanw) [inline templates/CRM/Admin/Form/ScheduleReminders.tpl:147]: `<nitpick>`
  I don't mean to be a pain... but I really don't like these comments. It's 2024 and we all know what a closing tag is.
`</nitpick> <!-- end of nitpick -->`

Comment (vingle) [reply templates/CRM/Admin/Form/ScheduleReminders.tpl:147]: Yes, especially as it sits next to the closing details tag (or should be). Perhaps at the end of this there can be a civi wide cleanup of `<!-- /.crm-accordion-wrapper -->` & `?

Comment (colemanw) [reply templates/CRM/Admin/Form/ScheduleReminders.tpl:147]: Sure. Ideally this PR wouldn't be adding new ones that we then have to cleanup.
But if that's easier then ok I can merge this.

Comment (aydun) [reply templates/CRM/Admin/Form/ScheduleReminders.tpl:147]: Those comments are now removed for all the templates in #29448

**Pull request merged**
